### DISCUSSION TITLE
ApiPlatform - Only validate the sent value

### DISCUSSION THREAD

**Discussion by ciastko1742** [symfony/symfony#43404] [Q&A] [answered]
hello,
i have a problem that i can't deal with.
I have an entity with many properties.

```
<?php

/**
 * @ORM\Entity(repositoryClass=TestRepository::class)
 *
 * @ApiResource(
 *     collectionOperations={
 *          "get",
 *          "post"={
 *              "security_post_denormalize"="is_granted('ROLE_ADMIN')"
 *          }
 *     },
 *     itemOperations={
 *          "get"={"security"="is_granted('view', object)"},
 *          "put"
 *     }
 * )
 */
class Test
{

    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255, unique=false)
     */
    private $name;

    /**
     * @ORM\Column(type="string", length=255, unique=false)
     * @Assert\Expression(
     *     "this.active===true"
     * )
     */
    private $status='Not started';

    /**
     * @ORM\Column(type="boolean")
     * @Assert\Expression(
     *     "this.accept===false and this.paid===false"
     * )
     */
    private $active;

    /**
     * @ORM\Column(type="boolean")
     * @Assert\Expression(
     *     "this.active===true"
     * )
     */
    private $accept;

    /**
     * @ORM\Column(type="boolean")
     * @Assert\Expression(
     *     "this.active===true"
     * )
     */
    private $paid;
}
```

Properties have added validation and are interdependent. For example, if "active === true" I can change "paid". All this is done with "Assert \ Expression". Now there is a problem when I want to change one property, e.g. $ name. I am sending a PUT with one value and the validation works on all properties. I know I can use validation groups, but how do I use them to distinguish the value sent? If I only send:
{
"name": "example name"
}
I don't want to validate other properties.
If my approach is wrong and it should be otherwise, give me a hint :)

**Comment by lyrixx** [comment, marked as answer]
Why don't you want to validate the whole object when only the name is sent? It's a code smell

Otherwise you can add more API endpoint, to only validate one property (with validation group) 

The other options would be to store what property has changed during the deserializion and use it during the the validation

**Comment by stof** [comment]
Also, for support about ApiPlatform, please ask it in the ApiPlatform support channels instead. You have a better chance of getting answers from the people knowing the answer.
